taipy/core/_repository/_fs_base.py

```diff
@@ -90,29 +90,44 @@ def __init__(self, model: Type[ModelType], dir_name: str):
     def dir_path(self):
         return self._storage_folder / self._dir_name
 
-    @property
-    def _directory(self) -> pathlib.Path:
-        return self.dir_path
-
     def load(self, model_id: str) -> Entity:
-        return self.__to_entity(self.__get_model_filepath(model_id))
+        try:
+            return self.__to_entity(self.__get_model_filepath(model_id))
+        except FileNotFoundError:
+            raise ModelNotFound(str(self.dir_path), model_id)
```

Comment:
since we're calling to_entity, I think EntityNotFound is a better name

Reply:
Good point. This is the historical implementation, but we could change it.

Reply:
@jrobinAV  what do you think?

Reply:
absolutely

Reply:
Let's go for EntityNotFound !

Reply:
We missed a part of the implementation.
We used ModelNotFound because we know nothing of the type of entity that we are searching for.
I think -- I don't remember -- that we design it for throwing "ModelNotFound" because technically we are searching for a Model. But this error should be handled by the caller layer that will return the correct error like "TaskNotFound".
What do you think?

Reply:
Absolutely agree :D

Reply:
On the layer that called the function, we are returning an error log (with the entity class name), I think this is good enough